src/HealthGPS/analysis_module.cpp

```diff
@@ -26,6 +26,41 @@ SimulationModuleType AnalysisModule::type() const noexcept {
     return SimulationModuleType::Analysis;
 }
 
+void AnalysisModule::initialise_vector(RuntimeContext &context) {
+    // Set some arbitrary factors
+    std::vector<core::Identifier> factors{"Gender"_id, "Age"_id};
+
+    std::vector<int> factor_bins;
```

Comment:
To avoid potential reallocations (unlikely with few keys, but still), one can `reserve` space inside the vector before pushing back.
```suggestion
    std::vector<int> factor_bins;
    factor_bins.reserve(factors.size());
```

Reply:
Sometimes `clang-tidy` will give this advice too, but it seems not to have done here for some reason